lib/fluent/plugin/in_watch_process.rb

```diff
@@ -44,53 +42,237 @@ def shutdown
+        end
+
+        def default_command
+          command = [
+            command_ps,
+            pipe_filtering_normal_ps,
+            pipe_select_columns,
+            pipe_fixing_locale,
+            pipe_formatting_output,
+          ].join
+          "powershell -command \"#{command}\""
+        end
+
+        private
+
+        def command_ps
+          if @keys.include?("user")
+            # The "IncludeUserName" option is needed to get the username, but this option requires Administrator privilege.
+            "Get-Process -IncludeUserName"
+          else
+            "Get-Process"
+          end
+        end
+
+        def pipe_filtering_normal_ps
+          # There are some special processes that don't have some properties, such as the "Idle" process.
+          # Normally, these are specific to the system and are not important, so exclude them.
+          # Note: The same situation can occur in some processes if there are no Administrator privilege.
+          " | ?{$_.StartTime -ne $NULL -and $_.CPU -ne $NULL}"
+        end
+
+        def pipe_select_columns
+          columns = @keys.map { |key|
+            next unless DEFAULT_PARAMS.keys.include?(key)
+            DEFAULT_PARAMS[key]
+          }.compact
+
+          if columns.nil? || columns.empty?
+            raise "The 'keys' parameter is not specified correctly. [keys: #{@keys}]"
+          end
+
+          " | Select-Object -Property #{columns.join(',')}"
+        end
+
+        def pipe_fixing_locale(format: "ddd MMM dd HH:mm:ss yyyy", locale: "en-US")
+          # In Windows, setting the "$env:Lang" environment variable is not effective in changing the format of the output.
+          # You can use "Datetime.ToString" method to change format of datetime values in for-each pipe.
+          # Note: About "DateTime.ToString" method: https://docs.microsoft.com/en-us/dotnet/api/system.datetime.tostring
+          # Note: About "Custom date and time format strings": https://docs.microsoft.com/en-us/dotnet/standard/base-types/custom-date-and-time-format-strings
+          unless @keys.include?("start_time")
+            return ""
+          end
+          " | %{
+            $_.StartTime = $_.StartTime.ToString(
+              '#{format}',
+              [Globalization.CultureInfo]::GetCultureInfo('#{locale}').DateTimeFormat
+            ); 
+            return $_;
+          }"
+        end
+
+        def pipe_formatting_output
+          # In the "ConvertTo-Csv" command, there are 2 lines of type info and header info at the beginning in the outputs.
+          # By the "NoTypeInformation" option, the line of type info is excluded.
+          # This enables you to skip the just first line for parsing, like linux or mac.
+          " | ConvertTo-Csv -NoTypeInformation"
+        end
+      end
+
+      module OS
+        # ref. http://stackoverflow.com/questions/170956/how-can-i-find-which-operating-system-my-ruby-program-is-running-on
+        def self.windows?
```

Comment:
Thank you for the advice.
I knew about it in `env.rb`, but was hesitant to change it because of the different definitions.
If it is better to change this, I will do!

Reply:
I have fixed this in https://github.com/daipom/fluent-plugin-watch-process/pull/2/commits/f8f4f8ddd6c2bab8afa8b1b24b70bba7d1aae5d5.